### ISSUE TITLE
TensorFlow build is failing on Bazel CI due to Eigen update

### ISSUE THREAD

**Issue by meteorcloudy** [tensorflow/tensorflow#39180]
https://buildkite.com/bazel/tensorflow/builds/4927
The build is failing on Linux and macOS with
```
./third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/PacketMathAVX512.h:432:16: error: could not convert 'Eigen::internal::pfirst<__vector(2) long long int>(_mm_min_epi32(res.Eigen::internal::eigen_packet_wrapper<__vector(2) long long int, 0>::operator __vector(2) long long int&(), _mm_shuffle_epi32(res.Eigen::internal::eigen_packet_wrapper<__vector(2) long long int, 0>::operator __vector(2) long long int&(), ((((0 << 6) | (0 << 4)) | (0 << 2)) | 1))))' from 'Eigen::internal::unpacket_traits<__vector(2) long long int>::type {aka __vector(2) long long int}' to 'Eigen::QInt32'
   return pfirst(
          ~~~~~~^
       _mm_min_epi32(res, _mm_shuffle_epi32(res, _MM_SHUFFLE(0, 0, 0, 1))));
```

It's weird we don't see the failure on TF CI.

A bisect shows 510f0f9a6c79d5e2f29c52911a3819fbdb61a6e0 is the culprit

/cc @gunan

**Comment by gunan** [comment]
I am puzzled, as all TF builds look healthy...
@mihaimaruseac have we seen this error?
@rmlarsen any ideas?

**Comment by rmlarsen** [comment]
I've seen a related issue where specific integer intrinsics were MIA from the standard header on some versions of GCC. I'll try to work around it in Eigen.

**Comment by mihaimaruseac** [comment]
I haven't seen it all on internal CI.

**Comment by rmlarsen** [comment]
@meteorcloudy how do I repro this build failure?

**Comment by rmlarsen** [comment]
I can try to update the old FixedPoint code in TF to use the eigen_packet_wrapper, but I'm not sure if that will fix it.

**Comment by rmlarsen** [comment]
I was able to repro in TF with -mavx2.

**Comment by rmlarsen** [comment]
I have a fix in cl/310035740.

**Comment by marvin-Yu** [comment]
Hi @rmlarsen, could you show the fix solution?
